multimodal/src/autogluon/multimodal/data/process_label.py

```diff
@@ -15,15 +16,15 @@ class LabelProcessor:
 
     def __init__(
         self,
-        prefix: str,
+        model: nn.Module,
```

Comment:
Ideally, this file should not depend on `torch` since it's only about data processing.

We should allow users to use classes in this file without torch.

Reply:
In AutoMM, data processors are created for models. In data processors, we may need to use model configs (e.g., timm, hugginface, mmdet). This logic seem reasonable. Besides, `torch` is a basic package in AutoMM, which is as basic as `numpy`. 

`process_text` and `process_image` are conditioned on the third-party packages like `timm`, `transformers`, `mmcv`, etc. We may move them into the model definition. In this way, the processor classes are clearer.

Reply:
Would it make sense to support both prefix and model as arguments to the class initializer?

```python
def __init__(
   self,
   prefix: str,
   model: nn.Module, 
   ...
)
```

I'm raising this concern because of the deployment scenario. Ideally, the user should be able to create these processors without knowing the model, and creating the model will incur additional overhead.

Reply:
If we consider the deployment scenario, we may ideally ensure the processors do not depend on the models. This is largely a two-way door though. We can refactor it in a later stage. Just need to bear it in mind.

Reply:
We can convert model to a deployable format like ONNX / torchscript. However, if we add another layer of dependency, the logic will be quite complicated.

Reply:
Thanks for the reminding. 

1. Only the processor creation depends on model. I guess we probably only need to load the processors in deployment. 
2. The current image and text processor require the packages timm, huggingface, mmcv, which I think are more complicated than the plain torch package. By using the model as input, we can further remove timm, huggingface, mmcv from processors.